Reorganize Ansible playbooks

conorsch opened the pull request:
Substantial changes to the flow of the Ansible playbooks and structure of the roles. Most of the diff is due to files that have been moved or renamed—functional changes were kept to a minimum in order to facilitate review. Once merged, future changes to the Ansible roles will be easier to make and to review.

### Major changes
* Consolidates host-specific shared roles (e.g. `common-app`, `common-mon` -> `common`) 
* Use `group_vars` for `staging` and `development` hosts
* Manage group membership in `manage-groups` role for production instances
* Manage group membership in `Vagrantfile` for testing VMs
* Support `ANSIBLE_ARGS` environment variable for managing tags and skipped tags on testing machines

### Minor changes
* More readable task names (also added task names for plays)
* Several improvements to idempotence (e.g. `command` calls only report `changed` when appropriate)
* Typo fixes (e.g. iptables comments)

Happy to review in person since the diff is large. Once this PR is merged, there's a list of **ops**-tagged issues outstanding that I plan to knock out posthaste.

ageis commented on install_files/ansible-base/roles/app/tasks/setup_cron.yml:
```diff
@@ -1,9 +1,8 @@
 ---
-- name: add cron job to cleanup SecureDrop temp dir daily
+- name: Add cronjob to clean SecureDrop tmp dir daily.
```
I still call it a cron job, not a cronjob ;)

conorsch replied:
Ha! No contest, I'll change it.

conorsch commented:
@garrettr @ageis Addressed the issues discussed during in-person review. Regarding the idempotence changes for `/etc/hosts`, the config now removes duplicate lines from the file, then performs the `lineinfile` substitution. Requires three tasks rather than one, but ensures idempotence and also cleans up after old, deprecated tasks.

Implemented the same three-task approach for cleaning up `/etc/default/haveged`, as well.
